bluepyopt/ephys/create_hoc.py

```diff
@@ -169,19 +160,57 @@ def create_hoc(
     else:
         banner = None
 
+    return dict(global_params=global_params, 
+                ignored_global_params=ignored_global_params,
+                section_params=section_params, 
+                range_params=range_params, 
+                location_order=location_order,
+                channels=channels,
+                banner=banner)
+
+
+def create_hoc(mechs,
+               parameters,
+               morphology=None,
+               ignored_globals=(),
+               replace_axon=None,
```

Comment:
We need to keep `replace_axon` in mind as a potential source of issues. I believe this is used often and we don't have a (simple) way to do that in Arbor yet.

Reply:
Yes, indeed, this is an important part of our models. I believe many network modelers will use a similar construct since simulating the full axon is mostly computationally wasteful. (But the full axon needs to be in the file to determine connections etc).

Reply:
Since there is currently no equivalent way to do this in Arbor, I'm [raising an exception](https://github.com/BlueBrain/BluePyOpt/pull/393/files#diff-ad07d90a013f2bddbcff4602145ad724e31e01491313b4a41ffc38359b53946dR320-R322) in `create_acc` if invoked with `replace_axon == True` (I've also added a test for that). To compare simulation outputs between Neuron and Arbor, there is an example in the [Allen tutorial PR](https://github.com/arbor-sim/arbor/pull/1781/files) with a figure under `doc/tutorial/single_cell_allen_result.svg` showing agreement when no axon replacement is being performed. I can prepare a similar example for BluePyOpt.

Reply:
Ok, but so just be aware that when this exception is not resolved, the usefulness of the code will be limited since all of our models use replace_axon, and as I mentioned, I see many network modelers wanting the same functionality

Reply:
Ok, I'm having a try at `replace_axon` for Arbor. The default policy seems to be implemented in the [morphologies module](https://github.com/BlueBrain/BluePyOpt/blob/master/bluepyopt/ephys/morphologies.py#L201-L246). I understand this code in the sense that two 30 um segments replace the axon in the morphology. I'm not entirely sure what the diameter (`diam`, `D1`, `D2`) evaluate to. When trying to recreate the voltage traces from the simple cell example, I'm getting very good overlap choosing a diameter of 1 um. Is this the right value (and if so, why)?

![Screenshot from 2022-07-21 21-46-19](https://user-images.githubusercontent.com/38319063/180304774-b1512d72-2b5e-49da-974b-2a202ecaa3ae.png)

Without axon replacement, the voltage traces are almost identical to the eye between Arbor and Neuron. I can share the results shortly. For the layer 5 pyramid cell, I'm getting this axon replacement (blue), by stretching the first segment of the original axon again to two 30 um segments (applying the scaling also to the radius). Is this the correct approach?

![Screenshot from 2022-07-21 16-04-15](https://user-images.githubusercontent.com/38319063/180305446-41e1814b-5d41-46e1-b101-347560e617f3.png)

Reply:
The D1 and D2 values normally represent the diameters of the original morphology

Reply:
For the simplecell there is no axon afaik, so it would default to a diameter of 1. It might be useful to compare your exact diameters for the L5PC example.

Reply:
To be fully consistent, I'm now using Neuron's diameters that can be read out after `cell.instantiate_morphology(sim)` is invoked. They do not correspond to the diameter of any individual segment, though, when the morphology is read in Arbor (my closest guess and default choice when the morphology is not instantiated with Neuron is therefore `median_distal_radii` at a certain distance from the soma).

Reply:
We need to handle the case when `section.nseg > 1` as well.

Reply:
Done in cde9b969a06da8d9cdecaeb4bd1ee7f84abf94df.